apps/ui/src/routes/index.ts

```diff
@@ -37,14 +38,17 @@ const router = createRouter({
 router.beforeEach((to, _from, next) => {
   let redirectPath: string | null = null;
 
-  // Match and redirect paths like "/safe.eth/settings" to "/s:safe.eth/settings"
-  const domainMatch = to.path.match(/^\/([^:\/]+?\.[^:\/]+)(\/.*)?$/);
-  if (domainMatch) {
-    const domain = domainMatch[1];
```

Comment:
Can't we just replace the domain here and not change the whole logic? There is plenty of case to test if we change the logic.

Reply:
Here we match only if the space doesn't have `s:` for example URLs like `/safe.eth/settings`, 

updated code handles  both  URLs with `s:safe.eth` and `safe.eth`  and redirects to new aliased space

Reply:
I'm a bit afraid of rewriting the logic like this, it might be hard to find out if we are breaking anything.

Do we need to redirect `safe.eth` to `s:safe.eth`? Space should have single canonical URL (aside from hardcoded redirects).

Reply:
> Do we need to redirect safe.eth to s:safe.eth

Yes, else all existing URLs will break, users reference these URLs on their forums or blogs or Twitter, etc.

Reply:
I have tested most of the cases (also mentioned them in the description)

Also, this is not a complete rewrite but adds one more case to handle aliases and moving `/about` to routes instead of handling it in this function

Reply:
I'm mostly concerned by whitelist functionality as this code was used for this as well.